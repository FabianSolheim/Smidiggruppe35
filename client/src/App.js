import React from "react"
import ReactDOM from "react-dom"
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import { ThemeProvider } from "@mui/material";
import theme from "./styling/theme"



export const App = () => {
    return(
        <>
        <ThemeProvider theme={theme}>
            <Typography color="text.primary" variant="h4" component="h2">
            Hello Meliora
            </Typography>
            <Button variant="contained" color="primary">
                Hello World
            </Button>
        </ThemeProvider>
        </>
    )
}


ReactDOM.render(<App/>, document.getElementById("root"))