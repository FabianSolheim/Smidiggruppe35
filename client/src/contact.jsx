import { useState } from "react";
import React from "react";
import Button from "@mui/material/Button";
import { TextField } from "@mui/material";

export function Contact() {
  return (
    <div>
      <h1>Contact</h1>
      <p>
        Reach out if you want to collaborate or know more about our services or
        just to connect
      </p>
        <ContactForm/>
    </div>
  );
}

function ContactForm() {
  const [name, setName] = useState("");
  const [companyName, setCompanyName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  return (
    <form>
      <div>
        <TextField required label={"Name"} variant={"standard"} />
          {/*
        <input
          id={"name"}
          name={name}
          onChange={(event) => setName(event.target.value)}
        />*/}
      </div>
      <div>
        <TextField required label={"Co. Name"} variant={"standard"} />
          {/*
        <input
          name={companyName}
          onChange={(event) => setCompanyName(event.target.value)}
        />*/}
      </div>
      <div>
        <TextField required label={"Email"} variant={"standard"}/>

          {/*
          <input
            name={email}
            onChange={(event) => setEmail(event.target.value)}
          />*/}
      </div>
      <div>
        <TextField required label="Message" multiline rows={4} />
          {/*
        <textarea
          name={message}
          onChange={(event) => setMessage(event.target.value)}
        />*/}
      </div>
      <div>
        <Button variant={"contained"}>Submit</Button>
      </div>
    </form>
  );
}
