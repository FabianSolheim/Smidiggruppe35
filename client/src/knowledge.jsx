import { useEffect, useState } from "react";
import "./styling/case.css";
import { Typography } from "@mui/material";
import { CaseStudyKnowledgeHeader } from "./pages/CaseStudyKnowledge/CaseStudyKnowledgeHeader";
import {HeadlineWithUnderline} from "./components/HeadlineWithUnderline/headlineWithUnderline";

function useLoading(loadingFunction) {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState();
  const [data, setData] = useState();

  async function load() {
    try {
      setLoading(true);
      setData(await loadingFunction());
    } catch (error) {
      setError(error);
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    load();
  }, []);

  return { loading, error, data };
}

export function KnowledgeCases() {
  const { loading, error, data } = useLoading(async () => {
    return fetchJSON("/api/knowledge");
  });

  if (loading) {
    return <div>Loading...</div>;
  }
  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div>{error.toString()}</div>
      </div>
    );
  }
  return (
    <div>
      <CaseStudyKnowledgeHeader />
      {HeadlineWithUnderline("Global Issue",175, "#034f7a")}
      <Typography id={"text"}>
        759 million adults are illiterate and do not have the awareness
        necessary to improve both their living conditions and those for their
        children. Without skills for lifelong learning, children face greater
        barriers to earning potential and employment later in life. They are
        more likely to suffer adverse health outcomes and less likely to
        participate in the decisions that affect them â€“ threatening their
        ability to build a better future for themselves and their communities.
      </Typography>
      <div className={"list"}>
        {data.map((knowledge) => (
          <KnowledgeCard key={knowledge.id} knowledge={knowledge} />
        ))}
      </div>
    </div>
  );
}

function KnowledgeCard({ knowledge: { name, info } }) {
  return (
    <>
      <div className={"knowledge"}>
        <Typography id={"headline"}>{name}</Typography>
        {info.map((info) => (
          <InfoCard key={info.id} info={info} />
        ))}
      </div>
    </>
  );
}

function InfoCard({ info: { title, description, imagetext, image } }) {
  return (
    <>
      {image && <img src={image} id={"image"} alt={"Image"} />}
      <Typography id={"imageText"}>{imagetext}</Typography>
      <Typography id={"head"}>{title}</Typography>
      <Typography>{description}</Typography>
    </>
  );
}

async function fetchJSON(url) {
  const res = await fetch(url);
  if (!res.ok) {
    throw new Error(`failed to load${res.status}: ${res.statusText}`);
  }
  return await res.json();
}
