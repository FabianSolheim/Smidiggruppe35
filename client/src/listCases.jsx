import {useEffect, useState} from "react";

function useLoading(loadingFunction) {

    const [loading, setLoading] = useState(true);
    const [error, setError] = useState();
    const [data, setData] = useState();

    async function load() {
        try {
            setLoading(true);
            setData(await loadingFunction());
        } catch (error) {
            setError(error);
        } finally {
            setLoading(false);
        }
    }

    useEffect(() => {
        load();
    }, []);

    return {loading, error, data};
}

export function ListCases() {
    const {loading, error, data} = useLoading(async () => {
        return fetchJSON("/api/cases")
    });

    if (loading) {
        return <div>Loading...</div>;
    }
    if (error) {
        return (
            <div>
                <h1>Error</h1>
                <div>{error.toString()}</div>
            </div>);
    }
    return (
        <div>
            {data.map((organization) => (
                <OrganizationCard key={organization._id} organizations={organization}/>
            ))}
        </div>);
}

function OrganizationCard({organizations: {name, category, info}}) {
    return <>
        <h1>{name}</h1>
        <h3>{category}</h3>
        {info.map((info) => (
            <InfoCard key={info._id} info={info}/>
        ))}
    </>
}

function InfoCard({info: {title, description, image}}) {
    return <>
        <h2>{title}</h2>
        <div>{description}</div>
        {image && <img src={image}/>}
    </>
}

async function fetchJSON(url) {
    const res = await fetch(url)
    if (!res.ok) {
        throw new Error(`failed to load${res.status}: ${res.statusText}`);
    }
    return await res.json();
}