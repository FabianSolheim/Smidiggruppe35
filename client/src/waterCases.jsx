import { useEffect, useState } from "react";
import "./styling/case.css";
import { Typography } from "@mui/material";
import {Underline} from "./components/Underline/underline";

function useLoading(loadingFunction) {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState();
  const [data, setData] = useState();

  async function load() {
    try {
      setLoading(true);
      setData(await loadingFunction());
    } catch (error) {
      setError(error);
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    load();
  }, []);

  return { loading, error, data };
}

export function WaterCases() {
  const { loading, error, data } = useLoading(async () => {
    return fetchJSON("/api/water");
  });

  if (loading) {
    return <div>Loading...</div>;
  }
  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div>{error.toString()}</div>
      </div>
    );
  }
  return (
    <div>
      <Typography id={"top"}>Limited access</Typography>
        {Underline(215, "#034f7a")}
      <Typography id={"text"}>
        Thatâ€™s nearly 1 in 10 worldwide. Or, twice the population of the United
        States. The majority live in isolated rural areas and spend hours every
        day walking to collect water for their family. Not only does walking for
        water keep children out of school or take up time that parents could be
        using to earn money, but the water often carries diseases that can make
        everyone sick. But access to clean water means education, income and
        health - especially for women and kids.
      </Typography>
      <div className={"list"}>
        {data.map((organization) => (
          <OrganizationCard
            key={organization._id}
            organizations={organization}
          />
        ))}
      </div>
    </div>
  );
}

function OrganizationCard({ organizations: { name, info } }) {
  return (
    <>
      <div className={"organization"}>
        <Typography id={"headline"}>{name}</Typography>
        {info.map((info) => (
          <InfoCard key={info._id} info={info} />
        ))}
      </div>
    </>
  );
}

function InfoCard({info: {title, description, imagetext, image}}) {
    return <>
        {image && <img src={image} id={"image"} alt={"Image"}/>}
        <Typography id={"imageText"}>{imagetext}</Typography>
        <Typography id={"head"}>{title}</Typography>
        <Typography>{description}</Typography>

    </>
}

async function fetchJSON(url) {
  const res = await fetch(url);
  if (!res.ok) {
    throw new Error(`failed to load${res.status}: ${res.statusText}`);
  }
  return await res.json();
}
