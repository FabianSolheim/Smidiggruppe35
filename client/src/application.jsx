import { BrowserRouter, Route, Routes } from "react-router-dom";
import React from "react";
import { Home } from "./pages/Home/Home";
import TestPage from "./pages/TestPage";
import { FaqPage } from "./pages/FaqPage/FaqPage";
import Navbar from "./components/Navbar/Navbar";
import { ThemeProvider } from "@mui/system";
import theme from "./styling/theme";
import { ListCases } from "./listCases";
import AnimationContainer from "./components/AnimationContainer/AnimationContainer";
import Header from "./components/navbar";
import { Contact } from "./pages/contact";
import { KnowledgeCases } from "./knowledge";
import PrivacyPage from "./pages/PrivacyPage/PrivacyPage";
import { AboutUsPage } from "./pages/AboutUsPage";
import { GetFaq } from "./pages/Faq/Faq";
import { CorporateMember } from "./pages/CorporateMember/corporateMember";
import { Footer } from "./components/Footer/footer";

export function Application() {
  return (
    <ThemeProvider theme={theme}>
      <AnimationContainer>
        <BrowserRouter>
          <header>
            <Navbar />
          </header>
          <Routes>
            <Route path={"/contact"} element={<Contact />} />
            <Route path={"/home"} element={<Home />} />
            <Route path={"/"} element={<CorporateMember />} />
            <Route path={"/test"} element={<TestPage />} />
            <Route path={"/faq"} element={<FaqPage />} />
            <Route path={"/privacy"} element={<PrivacyPage />} />
            <Route path={"/about"} element={<AboutUsPage />} />
          </Routes>
          <Footer />
        </BrowserRouter>
      </AnimationContainer>
    </ThemeProvider>
  );
}
