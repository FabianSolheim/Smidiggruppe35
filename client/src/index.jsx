import React from "react";
import * as ReactDom from "react-dom";
import { Application } from "./application";

ReactDom.render(<Application />, document.getElementById("root"));
