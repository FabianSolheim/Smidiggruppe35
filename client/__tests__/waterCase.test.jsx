import React from "react";
import ReactDOM from "react-dom";
import { WaterCases } from "../src/waterCases";

describe("WaterCase page", () => {
  it("shows waterCase page", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<WaterCases />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
