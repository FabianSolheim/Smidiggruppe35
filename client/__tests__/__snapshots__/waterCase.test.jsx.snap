// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaterCase page shows waterCase page 1`] = `"<div>Loading...</div>"`;
